t/501-hooks-yaml-metadata.t

```diff
@@ -0,0 +1,32 @@
+use strict;
+use warnings;
+use utf8;
+
+use Markdown::Perl 'convert', 'set_hooks';
+use Test2::V0;
+
+my $p = Markdown::Perl->new();
+my $page = <<EOF;
+---
+name: Mark is down
+draft: false
+number: 42
+---
+# Mark is down!
+
+I repeat: "Mark is down!"
+EOF
+
+# Test 1: Check if we can get a string value
+{
+  sub hook_is_name_mark {
+    my $x = shift;
+    ok(exists($x->[0]->{name}) && $x->[0]->{name} eq 'Mark is down', "key 'name' was retrieved and validated as being 'Mark is down'");
+  }
+  $p->set_hooks(yaml_metadata => \&hook_is_name_mark);
+  $p->convert($page);
+}
+
+# Test 2: Validate that hook is not called if yaml is invalid
```

Comment:
Are you missing this test?

Reply:
Oh yeah, i forgot to `git add` the test file.
I also tried adding tests for the `carp` behaviour with `Test::Carp` but somehow that causes a different codepath to be taken and does not give the expected result whatever I try. :( 

For example below, for unknown reasons to me, it does not call the hook (but if i run it as `$p->convert($page)` it does).
```perl
# Test 4: What happens if inside the hook we die()
use Test::Carp;

{
  sub hook_die {
    print "Dying ongoing...\n";
    die "We just die\n";
  }
  $p->set_hooks(yaml_metadata => \&hook_die);
  does_carp_that_matches($p->can("convert"), $page, qr/Hook.+crash/);
}
```

Reply:
I expect that this is because Test::Carp is incompatible with Test2 (so the hook is called but not caught by the test util). Instead you can use [Test2::Tools::Warnings](https://metacpan.org/pod/Test2::Tools::Warnings) (which I believe is already included automatically by Test2::V0.

Also it might be because I expect that the current code dies in the call to convert because it’s dereferencing an undef $blocks returned by _parse (this will problem will go away once the exception is just propagated to the caller).